docs/asf-docs/Introduction_hyp3_mintpy.ipynb

```diff
@@ -0,0 +1,527 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "hyp3 = sdk.HyP3(prompt = True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Submit the jobs with hyp3-sdk and download the files "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def run_job(hyp3, granule_list_file, projectname, outdir):\n",
+    "    #read the granulelist file\n",
+    "    with open(granule_list_file) as f:\n",
+    "        lines = f.read().splitlines()\n",
+    "\n",
+    "    num = len(lines)\n",
+    "    reference_id = lines[num-1]  \n",
+    "    insar_jobs = sdk.Batch()\n",
+    "    \n",
+    "    for i in range(num-2, 0, -1):\n",
+    "        secondary_id = lines[i]\n",
+    "        job_name = projectname\n",
+    "        insar_jobs += hyp3.submit_insar_job(reference_id, secondary_id, job_name,\n",
+    "                                    include_look_vectors=True, include_inc_map=True,\n",
+    "                                    include_los_displacement=True, include_dem=True,\n",
+    "                                    include_wrapped_phase=True)\n",
+    "\n",
+    "    #check and download the job\n",
+    "    if not insar_jobs.complete():\n",
+    "        # to get updated information\n",
+    "        insar_jobs = hyp3.refresh(insar_jobs)\n",
+    "        # or to wait until completion and get updated information (which will take a fair bit)\n",
+    "        insar_jobs = hyp3.watch(insar_jobs)\n",
+    "\n",
+    "    #download the jobs files\n",
+    "    os.system(\"mkdir -p {}\".format(outdir))\n",
+    "    os.chdir(outdir)\n",
+    "    insar_jobs.download_files()\n",
+    "    print(\"completed ...\")\n",
+    "\n",
+    "run_job(hyp3, granule_list, projectname, outdir)\n",
+    "print(\"downloaded the insar zip files are stored at {}.format(outdir)\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "2.2 Configure the custom cfg file in the outdir"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def config(template_file, data_dir, projectname):\n",
+    "    '''\n",
+    "    read in the template, produce the custom.txt and place inti inputs\n",
+    "    '''\n",
+    "    out_lines=[]\n",
+    "    with open(template_file) as file:\n",
+    "        lines = file.readlines()\n",
+    "\n",
+    "    for line in lines:\n",
+    "        out_lines.append(line.replace('{DATA_DIR}', data_dir))\n",
+    "        \n",
+    "    outfile = \"{}/{}.cfg\".format(data_dir, projectname)\n",
+    "    out_fid = open(outfile, 'w')\n",
+    "    out_fid.write(\"\".join(out_lines))\n",
+    "    return outfile\n",
+    "\n",
+    "template_file =\"/home/jzhu4/projects/work/hyp3/hyp3-mintpy/docs/hyp3_mintpy_template.cfg\"\n",
+    "config_file = config(template_file, outdir, projectname)"
```

Comment:
I think it would be simpler to include the config file contents here, rather than pulling them from an external file, so the user can directly see the configuration settings.  Something like:

```
from pathlib import Path

config = f'''
mintpy.load.processor        = hyp3
##---------interferogram datasets:
mintpy.load.unwFile          = {outdir}/*/*unw_phase_clip.tif
mintpy.load.corFile          = {outdir}/*/*corr_clip.tif
##---------geometry datasets:
mintpy.load.demFile          = {outdir}/DEM/dem_clip.tif
##---------incAngleFile:
mintpy.load.incAngleFile    =  {outdir}/*/*inc_map_clip.tif
'''

config_file = f'{projectname}.cfg'
Path(config_file).write_text(config)
```

Reply:
changed